handlers/objects.go

```diff
@@ -305,6 +321,305 @@ func (a *API) SearchObjects(params operations.SearchObjectsParams, principal *mo
 		WithAccessControlAllowOrigin("*")
 }
 
+// GetContainerObject handler that returns object (using container ID and object ID).
+func (a *API) GetContainerObject(params operations.GetContainerObjectParams, principal *models.Principal) middleware.Responder {
+	errorResponse := operations.NewGetContainerObjectBadRequest()
+	ctx := params.HTTPRequest.Context()
+
+	addr, err := parseAddress(params.ContainerID, params.ObjectID)
+	if err != nil {
+		resp := a.logAndGetErrorResponse("invalid address", err)
+		return errorResponse.WithPayload(resp)
+	}
+
+	var prm client.PrmObjectGet
+
+	if principal != nil {
+		btoken, err := getBearerTokenFromString(string(*principal))
+		if err != nil {
+			resp := a.logAndGetErrorResponse("get bearer token", err)
+			return errorResponse.WithPayload(resp)
+		}
+		attachBearer(&prm, btoken)
+	}
+
+	header, payloadReader, err := a.pool.ObjectGetInit(ctx, addr.Container(), addr.Object(), a.signer, prm)
+	if err != nil {
+		if isNotFoundError(err) {
+			resp := a.logAndGetErrorResponse("not found", err)
+			return operations.NewGetContainerObjectNotFound().WithPayload(resp)
+		}
+		resp := a.logAndGetErrorResponse("get object", err)
+		return errorResponse.WithPayload(resp)
+	}
+
+	payloadSize := header.PayloadSize()
+	res := operations.NewGetContainerObjectOK()
+
+	responder, contentType := a.setAttributes(res, payloadSize, params.ContainerID, params.ObjectID, header, params.HTTPRequest.URL.Query().Get("download"))
+	var payload io.ReadCloser = payloadReader
+	if len(contentType) == 0 {
+		if payloadSize > 0 {
+			// determine the Content-Type from the payload head
+			var payloadHead []byte
+
+			contentType, payloadHead, err = readContentType(payloadSize, func(uint64) (io.Reader, error) {
+				return payload, nil
+			})
+			if err != nil {
+				resp := a.logAndGetErrorResponse("invalid  ContentType", err)
+				return errorResponse.WithPayload(resp)
+			}
+
+			// reset payload reader since a part of the data has been read
+			var headReader io.Reader = bytes.NewReader(payloadHead)
+
+			if uint64(len(payloadHead)) == payloadSize { // otherwise, we've already read full payload
```

Comment:
```suggestion
			if uint64(len(payloadHead)) != payloadSize { // otherwise, we've already read full payload
```
sry i've mistaken in https://github.com/nspcc-dev/neofs-rest-gw/pull/114#discussion_r1420359909

Reply:
I haven't noticed that, sorry, but we would have found it in testcases, I believe. Fixed.